File: 0682/solution.java
import java.util.List;
import java.util.ArrayList;

class Solution {
    public int calPoints(String[] operations) {
        List<Integer> stack = new ArrayList<>();
        for (String c : operations) {
            switch (c) {
                case "+":
                    stack.add(stack.getLast() + stack.get(stack.size() - 2));
                    break;
                case "D":
                    stack.add(stack.getLast() * 2);
                    break;
                case "C":
                    stack.removeLast();
                    break;
                default:
                    stack.add(Integer.parseInt(c));
            }
        }
        int res = 0;
        for (int num : stack) {
            res += num;
        }
        return res;
    }
}

File: 0682/solution.ts
function calPoints(operations: string[]): number {
  const res: number[] = [];
  for (const op of operations) {
    switch (op) {
      case "+":
        res.push(res[res.length - 1] + res[res.length - 2]);
        break;
      case "D":
        res.push(res[res.length - 1] * 2);
        break;
      case "C":
        res.pop();
        break;
      default:
        res.push(parseInt(op));
    }
  }

  return res.reduce((acc, el) => acc + el, 0);
}
